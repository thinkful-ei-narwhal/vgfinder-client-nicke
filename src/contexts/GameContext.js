import React, { Component } from "react";

export const nullGame = {
  id: "",
  title: "",
  description: "",
  genre: "",
  rating: "",
  release_date: "",
  developer: "",
  trailer_url: "",
  image_url_box_art: "",
  image_url_two: "",
  image_url_three: "",
  image_url_four: "",
  image_url_five: "",
};

const GameContext = React.createContext({
  game: nullGame,
  setGame: () => {},
  clearGame: () => {},
});

export default GameContext;

export class GameProvider extends Component {
  state = {
    game: nullGame,
  };

  setGame = (game) => {
    this.setState({ game });
  };

  clearGame = () => {
    this.setGame(nullGame);
  };

  render() {
    const value = {
      game: this.state.game,
      setGame: this.setGame,
      clearGame: this.clearGame,
    };
    return (
      <GameContext.Provider value={value}>
        {this.props.children}
      </GameContext.Provider>
    );
  }
}
