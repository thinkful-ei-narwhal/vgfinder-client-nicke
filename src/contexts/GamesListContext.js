import React, { Component } from "react";

const GamesListContext = React.createContext({
  gamesList: [],
  setGamesList: () => {},
});
export default GamesListContext;

export class GamesListProvider extends Component {
  state = {
    gamesList: [],
  };

  setGamesList = (gamesList) => {
    this.setState({ gamesList });
  };

  render() {
    const value = {
      gamesList: this.state.gamesList,
      setGamesList: this.setGamesList,
    };
    return (
      <GamesListContext.Provider value={value}>
        {this.props.children}
      </GamesListContext.Provider>
    );
  }
}
