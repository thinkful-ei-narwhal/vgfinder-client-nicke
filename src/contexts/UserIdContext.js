import React, { Component } from "react";

const UserIdContext = React.createContext({
  userId: undefined,
  wishlist: [],
  userGames: [],
  setUserId: () => {},
  setUserWishlist: () => {},
  setUserIdWishlistAndGames: () => {},
});

export default UserIdContext;

export class UserIdContextProvider extends Component {
  state = {
    userId: undefined,
    wishlist: [],
    userGames: [],
  };

  setUserId = (userId) => {
    this.setState({ userId });
  };

  setUserIdWishlistAndGames = (userId, wishlist, userGames) => {
    this.setState({ userId, wishlist, userGames });
  };

  render() {
    const value = {
      userId: this.state.userId,
      wishlist: this.state.wishlist,
      userGames: this.state.userGames,
      setUserId: this.setUserId,
      setUserIdWishlistAndGames: this.setUserIdWishlistAndGames,
    };

    return (
      <UserIdContext.Provider value={value}>
        {this.props.children}
      </UserIdContext.Provider>
    );
  }
}
