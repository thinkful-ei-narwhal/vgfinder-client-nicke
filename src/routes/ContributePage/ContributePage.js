import "./ContributePage.css";
import React, { Component } from "react";
import Header from "./../../components/Header/Header";
import { Section } from "./../../components/Utils/Utils";
import PostGameForm from "./../../components/PostGameForm/PostGameForm";
import DeleteGameForm from "./../../components/DeleteGameForm/DeleteGameForm";
import GamesApiService from "../../services/games-api-service";
import GamesListContext from "../../contexts/GamesListContext";
import Alert from "./../../components/Alert/Alert";

export default class ContributePage extends Component {
  static contextType = GamesListContext;

  constructor(props) {
    super(props);
    this.state = {
      loaded: false,
      error: null,
    };
  }

  handleErrorClick = () => {
    this.setState({ error: null });
  };

  componentDidMount() {
    GamesApiService.getGames()
      .then((res) => {
        this.context.setGamesList(res);
        this.setState({ loaded: true });
      })
      .catch((err) => this.setState({ error: err.message }));
  }

  renderContribute() {
    const { error } = this.state;
    return (
      <>
        <Header pathName={this.props.location.pathname} />
        <Section className="">
          <div role="alert">
            {error && (
              <Alert message={error} handleErrorClick={this.handleErrorClick} />
            )}
          </div>
          <h1>Submit Indie Game</h1>
          <h3>Can't find your favorite indie title? Add your own</h3>
          <PostGameForm />
        </Section>
        <Section className="">
          <h1>Delete Indie Game</h1>
          <h3>
            Don't think a games is indie enough to be on this site? Delete it.
          </h3>
          <DeleteGameForm />
        </Section>
      </>
    );
  }

  render() {
    return <>{this.state.loaded ? this.renderContribute() : null}</>;
  }
}
