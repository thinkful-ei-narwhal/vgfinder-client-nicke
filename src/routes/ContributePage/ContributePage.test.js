import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ContributePage from './ContributePage'
import { GamesListProvider } from './../../contexts/GamesListContext'

describe(`Contribute Page component`, () => {
  it('renders a Contribute Page component', () => {
    const wrapper = shallow(
      <GamesListProvider>
        <ContributePage />
      </GamesListProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
