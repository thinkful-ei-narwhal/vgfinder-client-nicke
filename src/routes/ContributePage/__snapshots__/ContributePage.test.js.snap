// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contribute Page component renders a Contribute Page component 1`] = `
<ContextProvider
  value={
    Object {
      "gamesList": Array [],
      "setGamesList": [Function],
    }
  }
>
  <ContributePage />
</ContextProvider>
`;
